src/coreclr/src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1239,6 +1239,24 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_Vector128_AllBitsSet:
+        case NI_Vector256_AllBitsSet:
+        {
+            assert(op1 == nullptr);
+            if (varTypeIsFloating(baseType) && compiler->compSupports(InstructionSet_AVX))
+            {
+                // The immediate 8 means Equal (unordered, non-signaling)
+                // This is not available without VEX prefix.
+                emit->emitIns_SIMD_R_R_R_I(ins, attr, targetReg, targetReg, targetReg, 8);
+            }
+            else
+            {
+                assert(varTypeIsIntegral(baseType) || !compiler->compSupports(InstructionSet_AVX));
+                emit->emitIns_SIMD_R_R_R(ins, attr, targetReg, targetReg, targetReg);
```

Comment:
What is used for the comparison for `float/double` when AVX isn't supported? Based on the instruction table, this is `cmpps/cmppd` still; but I don't think that is correct?

Reply:
Ohhhh, good catch.

I wonder how we'd change it here, though? hard-code the instruction here?

Reply:
We would need to special-case one of the paths.

That's also possibly an interesting case for CSE... If the table gives a particular set of instructions but codegen can optionally special-case something further or treat it slightly differently, how should that be handled @briansull ?

Reply:
The ValueNumber & CSE phase only uses the table to determine if the result type needs to be an input when generating the ValueNumber for the node.

If for two trees that have the same operation (i.e. GT_MUL or the same HW intrinsic) and  all of its operands have the identical value numbers, then normally we would give the same value number.  

For GT_CAST we incorporate the castto type as an extra operand to the value number.

I noticed that we also needed to do this for some SIMD and HW instrinsic nodes.
I determined that safest and easiest way to to do this was to examine the table of instructions and always incorporate the result type when there were two or more different valid instructions listed for a SIMD or HW instrinsic node.

This process is used for x86 and x64, I believe that we need to be more conservative on ARM64, so we always include an extra result type operand.

As as long as the table has two or more different instructions we will be good.
It would be bad to list all the same instructions or an illegal instruction and then use hand code logic to decide on the instruction.   If you want to record that an entry relies upon hand coded logic I would recommend using a brkpt or nop instruction as a marker for this behavior in the table.  We can add a check for this and assume that different instructions could be generated.  I don't think that it should matter if AVX is supported or not when deciding if we need an extra result type operand.

Reply:
> I don't think that it should matter if AVX is supported or not when deciding if we need an extra result type operand.

This would be a case of, when not using `VEX` encoding (SSE-SSE4.2) all types would use the same instruction. But when using the `VEX` encoding (AVX+), `float`/`double` would use different instructions that are more efficient.

Reply:
Left a comment that float/double may use different instructions depending on the encoding available. Also hardcoded the instruction inside the integer / non-VEX path.